py/mpconfig.h

```diff
@@ -850,6 +850,13 @@ typedef double mp_float_t;
 #define MICROPY_MODULE_BUILTIN_INIT (MICROPY_CONFIG_ROM_LEVEL_AT_LEAST_EXTRA_FEATURES)
 #endif
 
+// Whether to allow built-in modules to have sub-packages (by making the
+// sub-package a member of their locals dict). The sub-package __name__ must
+// be foo.bar, and only top-level modules may have __init__.
```

Comment:
Does `__init__` here refer to a module `__init__()` method or an `__init__.py` file?

Reply:
That is a bit unfortunate. We make occasional use of that and would have to come up with a workaround. Probably just have to init all modules at `mp_init()` whether they are used or not.

Reply:
You should be able to have a C-level `__init__` on the top-level module (eg `foo`, or in your case `pybricks`), which initialises that module and all submodules of it.  That would run when `foo` is imported, or any submodule like `foo.bar`.

Probably the comment above should be improved to point this out clearly, that importing any submodule will initialise its top-level module, ie `import foo.bar` will call `foo`'s `__init__`.

Reply:
Updated the comment.